### PR TITLE
Add some more overloads into the C++ interface

### PR DESCRIPTION
**Pull request by antiagainst**
The user may not be able to control the input; it's performance defect to force making a `vector` from C array in cases.

Depends on #424.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,6 +195,12 @@ There are three main entry points into the library.
   for the header and each parsed instruction.  The disassembler is implemented
   as a client of `spvBinaryParse`.
 * `spvValidate` implements the validator functionality. *Incomplete*
+* `spvValidateBinary` implements the validator functionality. *Incomplete*
+
+The C++ interface is comprised of two classes, `SpirvTools` and `Optimizer`,
+both in the `spvtools` namespace.
+* `SpirvTools` provides `Assemble`, `Disassemble`, and `Validate` methods.
+* `Optimizer` provides the `Optimize` method.
```
Comment (dneto0) [inline README.md]: This is now "Run"

File: include/spirv-tools/libspirv.hpp [hunk 2]
```diff
@@ -82,17 +82,26 @@ class SpirvTools {
   // Returns true on successful assembling. |binary| will be kept untouched if
   // assembling is unsuccessful.
   bool Assemble(const std::string& text, std::vector<uint32_t>* binary) const;
+  // |text_size| specifies the number of bytes in |text|.
+  bool Assemble(const char* text, size_t text_size,
```
Comment (dneto0) [inline include/spirv-tools/libspirv.hpp]: I think we don't need the terminating Null. That's unusual and should be stated here.

Comment (antiagainst) [inline include/spirv-tools/libspirv.hpp]: Added comments.

Actually this is to be consistent with `spvTextToBinary`, which takes a size parameter. Should we do normal null-terminating instead? It's kinda safe to require a size, but less conventional.

**Comment by dneto0** [comment]
+2

**Pull request merged**